### DISCUSSION TITLE
Can't able to train the num_classes=1

### DISCUSSION THREAD

**Discussion by kethan52** [torchgeo/torchgeo#599] [General]
```python
model =SemanticSegmentationTask(
    segmentation_model="fcn",
    encoder_name="resnet18",
    encoder_weights="imagenet",
    in_channels=9,
    num_classes=1,
    num_filters=128,
    loss="jaccard",
    ignore_zeros=True,
    learning_rate=0.1,
    learning_rate_schedule_patience=0.05,
    
)

trainer = pl.Trainer(

                  gpus=0,
                  callbacks=[checkpoint_callback, early_stopping_callback],
                  logger=[csv_logger],
                  min_epochs=10,
                  max_epochs=100,
                  precision=32,
                  log_every_n_steps=0.01,
                  max_steps=8,
                  
               )

trainer.fit(model,dl,dl)
```

Hi, as I want to do the binary sematic segmentation I need num_clasess=1 can I have any scope of doing with single class

**Comment by isaaccorley** [comment]
This seems to be a duplicate of your other issues #513 and #514. Can you add some context to why this is a separate issue?

**Comment by calebrob6** [comment]
`num_classes=1` will need to be trained with a different loss than PyTorch's cross entropy, which isn't currently implemented in `SemanticSegmentationTask` (you would need to use this https://pytorch.org/docs/stable/generated/torch.nn.BCELoss.html#torch.nn.BCELoss). Maybe we should add an assert that `num_classes >= 2`?

**Comment by adamjstewart** [comment]
Could we add a conditional that automatically switches to BCE for num_classes=1?

**Reply by geografif** [reply]
I'm just curious if it has been implemented. Also, for binary task (target object and background), should I use num_classes=1 or num_classes=2? @adamjstewart @calebrob6

**Reply by adamjstewart** [reply]
See https://github.com/microsoft/torchgeo/pull/2219! This should be merged and released soon. For now, you can use `num_classes=2`, but once this is merged you can use `task='binary'` instead. Also be sure to use `loss='bce'`.

**Reply by geografif** [reply]
@adamjstewart Thanks for the response!

**Comment by calebrob6** [comment]
Yeah probably, I just don't have time to play around with it :)

**Comment by adamjstewart** [comment]
I think this is also related to #245. So it's not just loss, it's also IoU.
